repos/system_upgrade/el8toel9/actors/checkcustomnetworkscripts/libraries/customnetworkscripts.py

```diff
@@ -0,0 +1,51 @@
+import os
+
+from leapp import reporting
+
+CUSTOM_NETWORK_SCRIPTS = [
+    "/sbin/ifup-local",
+    "/sbin/ifup-pre-local",
+    "/sbin/ifdown-local",
+    "/sbin/ifdown-pre-local",
+]
+
+
+def generate_report():
+    """ Generate reports informing user about possible manual intervention requirements """
+
+    DOC_URL = "https://red.ht/upgrading-RHEL-8-to-RHEL-9-network-scripts"
+
+    # Show documentation url if custom network-scripts detected
+    title = "custom network-scripts detected"
+    summary = "RHEL 9 does not support the legacy network-scripts package that was \
+deprecated in RHEL 8, and the custom network-scripts have been detected. In order to \
+proceed with the upgrade, you can move the custom network-scripts, for example, to \
+the `/opt/network-scripts/` directory."
+
+    reporting.create_report(
+        [
+            reporting.Title(title),
+            reporting.Summary(summary),
+            reporting.Remediation(
+                hint=(
+                    "Migrate the custom network-scripts to NetworkManager dispatcher"
+                    " scripts manually before the ugprade. Follow instructions in the"
+                    " official documentation."
+                )
+            ),
+            reporting.Severity(reporting.Severity.HIGH),
+            reporting.Flags([reporting.Flags.INHIBITOR]),
+            reporting.Tags([reporting.Tags.SERVICES]),
+            reporting.ExternalLink(
+                title="Upgrading from RHEL 8 to 9 - migrating custom network-scripts \
+to NetworkManager dispatcher scripts",
+                url=DOC_URL,
+            ),
+        ]
+        + [reporting.RelatedResource("file", file) for file in CUSTOM_NETWORK_SCRIPTS]
```

Comment:
@liangwen12year https://leapp.readthedocs.io/en/latest/python-coding-guidelines.html#avoid-shadowing-python-builtins

Do not use the `file` varname. You can use e.g. `fname`, `fpath`, `filename`, ...

Reply:
Thanks for commenting, updated.